velox/connectors/Connector.h

```diff
@@ -89,10 +91,19 @@ class ConnectorInsertTableHandle {
   }
 };
 
+using WriterParametersVector =
+    std::vector<std::shared_ptr<const WriterParameters>>;
+
 class DataSink {
  public:
   virtual ~DataSink() = default;
 
+  // Get all parameters used by the underlying writers.
+  virtual std::shared_ptr<const WriterParametersVector> getWriterParameters()
```

Comment:
Removed default implementation

Regarding a vector of WriteParameters, we would need multiple underlying writers inside a HiveDataSink to support partitions, and thus there're possibly multi WriteParameters for a DataSink. The DataSink needs to expose them so that TableWriter can wrap them and pass them to the next operator.